src/cwrapper.h

```diff
@@ -0,0 +1,98 @@
+#ifndef CWRAPPER_H
+#define CWRAPPER_H
+
+#include <gmp.h>
+
+#ifdef __cplusplus
+#include "basic.h"
+#include "constants.h"
+namespace CSymPy {
+    class CWrapper {
+        public:
+            RCP<const Basic> ptr;
+            CWrapper() {
+                ptr = zero;
+            }
+    };
+}
+using CSymPy::CWrapper;
+
+extern "C" {
+
+#else
+typedef struct CWrapper CWrapper;
+#endif
```

Comment:
Why do we need to distinguish if this file is being compiled with C++ or C? And why do we need to define the full class `CWrapper` in the C++ case? Cannot we move the definition of CWrapper into the .cpp file? So that we have just one header file, identical with C and C++?